clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -240,9 +237,10 @@ protected boolean flush(ByteBuffer buf) throws IOException {
     */
     @Override
     public void handshake() throws IOException {
-        // Reset state to support renegotiation. This can be removed if renegotiation support is removed.
         if (state == State.READY)
-            state = State.HANDSHAKE;
+            throw renegotiationException();
+        if (state == State.CLOSING)
+            throw closingException();
```

Comment:
This throws a different exception. Should we throw the same one?

Reply:
I think at the moment, we should never get here, so `IllegalStateException` is fine.